src/tiramisu_function.cpp

```diff
@@ -1634,7 +1740,19 @@ void tiramisu::function::lift_mpi_comp(tiramisu::computation *comp) {
     }
 }
 
-void tiramisu::function::codegen(const std::vector<tiramisu::buffer *> &arguments, const std::string obj_filename, const bool gen_cuda_stmt) {
+void tiramisu::function::codegen(const std::vector<tiramisu::buffer *> &arguments, const std::string obj_filename, const bool gen_cuda_stmt) 
+{
+    if (gen_cuda_stmt) 
+    {
+	if(!this->mapping.empty())
+	{
+		tiramisu::computation* c1 = this->get_first_cpt();
+		tiramisu::computation* c2 = this->get_last_cpt();
+		Automatic_communication(c1,c2);
+        }
+	else 
+	     	DEBUG(3, tiramisu::str_dump("You must specify the corresponding CPU buffer to each GPU buffer else you should do the communication manually"));    
```

Comment:
To be aligned with the rest of Tiramisu code, I prefer if you add
"set_atuomatic_gpu_copies(bool)"

By default, automatic copies are set to true. The user can disable it by calling this and setting it to false.

This should be similar to

tiramisu::buffer::set_auto_allocate(bool)

and

tiramisu::global::set_auto_data_mapping(bool)

The user expects gpu/host copies to be tunable with a similar API.

In this case you have to check whether the user set automatic copies to true but this->mapping.empty() is true. In that case you should signal an error (not just debugging). Errors are signaled using the ERROR() macro.

Reply:
"set_atuomatic_gpu_copies(bool)"  to say that i'll do or not copies automatically 
and this tiramisu::global::set_auto_data_mapping(bool) , is supposed to do what?

Reply:
ok

Reply:
1. here i preferred to use atuomatic_gpu_copies per buffer so that if the user wants to do a manual copy for a given buffer this will be possible,  
 2. for the  default computation, 
i didn't understand well why we do for example A.store_in(&A_gpu) instead of A.store_in(&A_cpu) alors que normalement on les données doivent passer par les buffers cpu , puis on fait une copie aux buffers gpu.
genre je ne vois pas le rôle de ici

Reply:
Sorry I didn't suggest that you add tiramisu::global::set_auto_data_mapping(bool). It already exists. I just gave you an example. You don't need to do anything for tiramisu::global::set_auto_data_mapping(bool).